tests/test_host_check.py

```diff
@@ -711,34 +851,34 @@ class TestArch(_CheckTestBase):
 
     def test_success(self, capsys):
         """Test the architecture being correct."""
-        success, output = self.perform_check(capsys, cmd_effects="x86_64")
-        assert output == "PASS -- CPU architecture (x86_64)\n"
-        assert success
+        result, output = self.perform_check(capsys, cmd_effects="x86_64")
+        assert textwrap.dedent(output) == "PASS -- CPU architecture (x86_64)\n"
+        assert result.is_success()
 
     def test_incorrect_arch(self, capsys):
         """Test the incorrect architecture case."""
-        success, output = self.perform_check(capsys, cmd_effects="arm64")
-        assert output == textwrap.dedent(
+        result, output = self.perform_check(capsys, cmd_effects="arm64")
+        assert textwrap.dedent(output) == textwrap.dedent(
             f"""\
-            FAIL -- CPU architecture
-                    The CPU architecture is arm64, but XRd only supports: x86_64.
+             FAIL -- CPU architecture
+                     The CPU architecture is arm64, but XRd only supports: x86_64.
             """
         )
-        assert not success
+        assert result.is_failure()
```

Comment:
I think explicitly doing `result is CheckState.FAILED` would be better (similar for the success cases)